Packages/com.unity.inputsystem/InputSystem/Plugins/InputForUI/InputSystemProvider.cs

```diff
@@ -0,0 +1,679 @@
+
+            DirectionNavigation(currentTime);
+
+            foreach (var ev in m_Events)
+            {
+                // we need to ignore some pointer events based on priority (touch->pen->mouse)
+                // this is mostly used to filter out simulated input, e.g. when pen is active it also generates mouse input
+                if (m_SeenTouchEvents && ev.type == Event.Type.PointerEvent && ev.eventSource == EventSource.Pen)
+                    m_PenState.Reset();
+                else if ((m_SeenTouchEvents || m_SeenPenEvents) &&
+                         ev.type == Event.Type.PointerEvent && (ev.eventSource == EventSource.Mouse || ev.eventSource == EventSource.Unspecified))
+                    m_MouseState.Reset();
+                else
+                    EventProvider.Dispatch(ev);
+            }
+
+            // Sometimes single lower priority events can be received when using Touch or Pen, on a different frame.
+            // To avoid dispatching them, the seen event flags aren't reset in between calls to OnPointerPerformed.
+            // Essentially, if we're moving with Touch or Pen, lower priority events aren't dispatch as well.
+            // Once OnClickPerformed is called, the seen flags are reset
+            if (m_ResetSeenEventsOnUpdate)
+            {
+                ResetSeenEvents();
+                m_ResetSeenEventsOnUpdate = false;
+            }
+
+            m_Events.Clear();
+        }
+
+        void ResetSeenEvents()
+        {
+            m_SeenTouchEvents = false;
+            m_SeenPenEvents = false;
+        }
+
+        //TODO: Refactor as there is no need for having almost the same implementation in the IM and ISX?
+        void DirectionNavigation(DiscreteTime currentTime)
+        {
+            var(move, axesButtonWerePressed) = ReadCurrentNavigationMoveVector();
+            var direction = NavigationEvent.DetermineMoveDirection(move);
+
+            // Checks for next/previous directions if no movement was detected
+            if (direction == NavigationEvent.Direction.None)
+            {
+                direction = ReadNextPreviousDirection();
+                axesButtonWerePressed = m_NextPreviousAction.WasPressedThisFrame();
+            }
+
+            if (direction == NavigationEvent.Direction.None)
+            {
+                m_RepeatHelper.Reset();
+            }
+            else
+            {
+                if (m_RepeatHelper.ShouldSendMoveEvent(currentTime, direction, axesButtonWerePressed))
+                {
+                    EventProvider.Dispatch(Event.From(new NavigationEvent
+                    {
+                        type = NavigationEvent.Type.Move,
+                        direction = direction,
+                        timestamp = currentTime,
+                        eventSource = GetEventSource(GetActiveDeviceFromDirection(direction)),
+                        playerId = k_DefaultPlayerId,
```

Comment:
I see this is a fixed value at the moment so will need updating later but fine for this PR